File: programers_Lv1/점프와 순간 이동.py
def solution(n):
    ans = 1
    div_n = n
    
    while div_n > 1:
        if div_n % 2 != 0:
            div_n = div_n // 2
            ans += 1
        else:
            div_n = div_n / 2

    return ans

print(solution(5)) # 2
print(solution(6)) # 2
print(solution(5000)) # 5

File: 점프와 순간 이동.js
function solution(n) {
  var ans = 1;
  let divN = n;

  while (divN > 1) {
    if (divN % 2 !== 0) {
      divN = parseInt(divN / 2);
      ans++;
    } else {
      divN = divN / 2;
    }
  }

  return ans;
}

console.log(solution(5)); // 2
console.log(solution(6)); // 2
console.log(solution(5000)); // 5
